test/ProtocolFeesImplementation.t.sol

```diff
@@ -218,4 +219,30 @@ contract ProtocolFeesTest is Test, GasSnapshot, Deployers {
         assertFalse(success);
         assertEq(protocolFee, 0);
     }
+
+    function test_fetchProtocolFee_gasLimit() public {
+        gasLimitFeeController = new GasLimitProtocolFeeControllerTest();
+        protocolFees.setProtocolFeeController(gasLimitFeeController);
+        vm.prank(address(gasLimitFeeController));
+        (bool success, uint24 protocolFee) = protocolFees.fetchProtocolFee(key);
+        assertFalse(success);
+        assertEq(protocolFee, 0);
+    }
+
+    function test_fetchProtocolFee_revertsWithProtocolFeeCannotBeFetched() public {
+        protocolFees = new ProtocolFeesImplementation(9079256829993496519);
```

Comment:
Would be great to have this as a constant or something

this file can have a `GAS_LIMIT` and then set it as an immutable in the implementation contract. And then that contract can just access it too?

Reply:
```
    function test_fetchProtocolFee_revertsWithProtocolFeeCannotBeFetched() public {
        uint256 gasLimit = 5000;
        protocolFees = new ProtocolFeesImplementation(gasLimit);
        protocolFees.setProtocolFeeController(feeController);
        vm.prank(address(feeController));
        vm.expectRevert(IProtocolFees.ProtocolFeeCannotBeFetched.selector);
        protocolFees.fetchProtocolFee{gas: gasLimit - 1}(key);
    }
```
I'm suggesting this because I'm concerned that the current method might increase the testing time.
Thank you for considering my suggestion.